### PR TITLE
feat: official -> peacock progression transfer

### PR DESCRIPTION
**Pull request by AnthonyFuller**
As title. Thank's to @vitphire's fork which had an early version of progression transfer, it was used for reference.

This PR updates the latest profile version to version 2. This restructures how we store action xp for sublocations and also fixes the player profile page.

### CODE REVIEW

**Comment by AnthonyFuller** [comment]
This PR is currently awaiting web ui changes by @RDIL.

File: components/webFeatures.ts [hunk 1]
```diff
@@ -217,4 +247,344 @@ webFeaturesRouter.get(
...
+                {
+                    id: req.query.user,
+                    extensions: [
+                        "achievements",
+                        "friends",
+                        "gameclient",
+                        "gamepersistentdata",
+                        "opportunityprogression",
+                        "progression",
+                    ],
```
Comment (grappigegovert) [inline components/webFeatures.ts:407]: Maybe also get defaultloadout ?

Comment (AnthonyFuller) [reply components/webFeatures.ts:407]: Good point, I just ignored that because loadout profiles. One moment please.

**Review by RDIL — APPROVED** [review]
RSLGTM

**Pull request merged**